### PR TITLE
GOVSI-628 - Add update email message to queue and send via Notify

### PR DESCRIPTION
**Pull request by JHjava**
## What?

 - Add update email message to queue and send via Notify
- Create a new template ID to send a confirmation email that the email has been updated
- Create a new lambda role for access to both dynamo and SQS
- Create a new notification type for when an EMAIL has been updated

## Why?

- When an update email request has been received we need to update the email in dynamodb and then put it in on an SQS queue so it can be sent via Notify.

**TODO**
- Validate the OTP

### CODE REVIEW

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdateEmailHandler.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        LOGGER.info(
+                "Authorizer parameters received: {}", input.getRequestContext().getAuthorizer());
+        try {
+            UpdateEmailRequest updateInfoRequest =
+                    objectMapper.readValue(input.getBody(), UpdateEmailRequest.class);
+            Subject subjectFromEmail =
+                    dynamoService.getSubjectFromEmail(updateInfoRequest.getExistingEmailAddress());
+            Map<String, Object> authorizerParams = input.getRequestContext().getAuthorizer();
+
+            if (!authorizerParams.containsKey("principalId")) {
```
Comment (sanisc) [inline account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdateEmailHandler.java:61]: there is a helper method for this that @cj1993 has added which is probably worth while using

File: shared/src/main/java/uk/gov/di/authentication/shared/entity/NotificationType.java [hunk 2]
```diff
@@ -3,5 +3,6 @@
 public enum NotificationType {
     VERIFY_EMAIL,
     VERIFY_PHONE_NUMBER,
-    MFA_SMS
+    MFA_SMS,
+    EMAIL_UPDATED
 }
```
Comment (sanisc) [inline shared/src/main/java/uk/gov/di/authentication/shared/entity/NotificationType.java:8]: its probably worth separating these out to use its own for account management even if they are the same

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdateEmailHandler.java (hunk 1 above)
Comment (JHjava) [reply account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdateEmailHandler.java:61]: Going to leave it as it is for now

File: shared/src/main/java/uk/gov/di/authentication/shared/entity/NotificationType.java (hunk 2 above)
Comment (JHjava) [reply shared/src/main/java/uk/gov/di/authentication/shared/entity/NotificationType.java:8]: Have updated

**Review by sanisc — DISMISSED** [review]

**Review by sanisc — APPROVED** [review]

**Pull request merged**